lib/webdriverLogger.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as stream from 'stream';
+
+import {CommandName, WebDriverCommand} from './webdriverCommands';
+
+export class WebDriverLogger {
+  logStream: stream.Writable;
+
+  public setLogDir(logDir: string) {
+    let logName = `webdriver_log_${process.pid}.txt`;
+    this.logStream = fs.createWriteStream(path.join(logDir, logName), {flags: 'a'});
+  }
+
+  printCommand(command: WebDriverCommand) {
+    switch (command.commandName) {
+      case CommandName.NewSession:
+        let desired = command.data['desiredCapabilities'];
+        return `Getting new "${desired['browserName']}" session`;
+      case CommandName.DeleteSession:
+        let sessionId = command.getParam('sessionId').slice(0, 6);
+        return `Deleting session ${sessionId}`;
+      case CommandName.Go:
+        return `Navigating to ${command.data['url']}`;
+      case CommandName.GetCurrentURL:
+        return `Getting current URL`;
+      default:
+        return `Unknown command ${command.data['url']}`;
+    }
+  }
+
+  timestamp(): string {
+    let d = new Date();
+    let hours = d.getHours() < 10 ? '0' + d.getHours() : d.getHours();
+    let minutes = d.getMinutes() < 10 ? '0' + d.getMinutes() : d.getMinutes();
+    let seconds = d.getSeconds() < 10 ? '0' + d.getSeconds() : d.getSeconds();
+    let millis = d.getMilliseconds().toString();
+    millis = '000'.slice(0, 3 - millis.length) + millis;
+    return `[${hours}:${minutes}:${seconds}.${millis}]`;
+  }
+
+  public logWebDriverCommand(command: WebDriverCommand) {
+    let cmdLog = this.printCommand(command);
+
+    let logLine: string;
+    if (command.getParam('sessionId')) {
+      let session = command.getParam('sessionId').slice(0, 6);
+      logLine = `${this.timestamp()} [${session}] ${cmdLog}\n`;
+    } else {
+      logLine = `${this.timestamp()} ${cmdLog}\n`;
+    }
+
+    this.logStream.write(logLine);
+  }
+}
```

Comment:
We should also log the duration of each command - and consider how we might log the start of a command which doesn't return (for debugging).